### DISCUSSION TITLE
Is there any intention of using the react option for parameters

### DISCUSSION THREAD

**Discussion by jwoLondon** [vega/vega-lite#9440] [Q&A] [answered]
Apologies if I am missing something obvious here, but I note in the [release V5.20.0 documentation](https://github.com/vega/vega-lite/releases/tag/v5.20.0), that a `react` is exposed as part of the parameter options, with the use-case being for interactive legends that filter data. Yet this doesn't seem to be part of the schema, nor have any effect in the editor (5.20.1), nor feature in the documentation. Is there supposed to be a `react` option, or should we use conditional encoding rather than filtering for the interactive legend use-case?

**Comment by mattijn** [comment]
We (esp @joelostblom and @dwootton) at the Altair repository, are exploring how we can use this `react` option. See eg this [Vega Editor URL](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEySYUqUAwBOgtBmx5CBKMtlSIyuMOLbKsuJgKycCXhX6r1ERxAACxACMlABMAJyUAAwEXkgmSNoQFnIgAL5KOAlICBBoANqgstlw+gBebGzITIJwAPrQSDW5Sj41UJhooJgAnjgl6DSymHDaxI2KIHCyUGzKg2S5qHkgAB4T3SAAukpQNQn6ytW7NFAA1mlKTIPK+g1NF4XF+jOCbNr11nDt7PItn+2dEA9Pr6HBsQYdJRTGZzWQLfIgF5vLZKFKaEDIbSnIoIfogADy2hotFktRqZCmyjJSCY1maQN6eOOZwuiL22gOR2ELPSICushu6PJlMBmOxT3QhOJgzJ5Ep1NpgnpwKZ3POqQ1CkeuOelRwDGGVOUlSQgwmcBWOA56GNyEGAAoAORI7SOgCUDxAOKZbFe71tpt+IG0cCQANQADNGj4oZbrYi9Qa4EaTWbUtsgdokLIIBG3gh8qAIzRBMNraBMlmC+gXR82pgfml00oxYCVaDwUMJsXBDVBZhtAw4EpMBVSzQcGgB0PedDZvNAS7AcXrIKCUSSRN2+hZJVBuMWlBGv1QAGzeiLVbdX7amf5BreWt0SvBGuABJvHxggDuIy3jP0ABHBhswbdQG1IVkNmfGhV30ABZEs4AgWo+neABxRpXiDbcQGA0C6DUGhIIfIA) from this comment https://github.com/vega/altair/pull/3394#issuecomment-2316286932.

**Reply by jwoLondon** [reply]
Thanks for this. 

Am I right in thinking that the only legitimate context in which `react` could be used is in constructing a new parameter based on an existing expression (that becomes the 'frozen' upstream value)?

So this is legitimate:

```json
"params": [
  {"name": "interactiveLegend", "select": {"type": "point", "encodings": ["color"]}, "bind": "legend" },
  {"name": "clr", "expr": "domain('color')"},
  {"name": "clrDomain", "react": false, "expr": "clr"}
]
```

but this would not be:

```json
"params": [
  {
    "name": "interactiveLegend",
    "select": {"type": "point", "encodings": ["color"]},
    "bind": "legend",
    "react": false,
  }
]
```

And further, there would never be a use case where `"react":true` would be needed?

**Reply by joelostblom** [reply, marked as answer]
> that becomes the 'frozen' upstream value

I think this would be the only usecase; at least the only one I had in mind when adding that param.

> there would never be a use case where "react":true would be needed?

I don't think so, that should be the default for all params.
